lingua_franca/format.py

```diff
@@ -242,7 +242,60 @@ def year_format(self, dt, lang, bc):
 
 
 date_time_format = DateTimeFormat(os.path.join(os.path.dirname(__file__),
-                                  'res/text'))
+                                               'res/text'))
+
+
+def nice_bytes(number, lang=None, speech=True, binary=True, gnu=False, places=1):
+    """
+    turns a number of bytes into a string using appropriate units
+
+    prefixes - https://en.wikipedia.org/wiki/Binary_prefix
+    spoken binary units - https://en.wikipedia.org/wiki/Kibibyte
+    implementation - http://stackoverflow.com/a/1094933/2444609
+
+    :param number: number of bytes (int)
+    :param lang: lang_code, ignored for now (str)
+    :param speech: spoken form (True) or short units (False)
+    :param binary: 1 kilobyte = 1024 bytes (True) or 1 kilobyte = 1000 bytes (False)
+    :param gnu: say only order of magnitude (bool)  - 100 Kilo (True) or 100 Kilobytes (False)
+    :param places: decimal places (int), TODO not yet implemented
+    :return: nice bytes (str)
+    """
+    lang_code = get_primary_lang_code(lang)
+
+    if speech and gnu:
+        default_units = ['Bytes', 'Kilo', 'Mega', 'Giga', 'Tera', 'Peta', 'Exa', 'Zetta', 'Yotta']
+    elif speech and binary:
+        default_units = ['Bytes', 'Kibibytes', 'Mebibytes', 'Gibibytes',
+                         'Tebibytes', 'Pebibytes', 'Exbibytes', 'Zebibytes', 'Yobibytes']
```

Comment:
I think we should check this in a couple different TTS engines, to make sure Mycroft can actually say "kibibytes" and "exbibytes"

Reply:
if a TTS engine cant read that is out of scope from LF, in the end this repo is about formatting and parsing, TTS doesnt even come into the picture until downstream uses it

Mimic2 has a mechanism for users to correct pronunciations, something like that could be expanded for other engines, but this is very implementation specific and out of scope not only of this PR but of this repo

Reply:
Yeah. I was thinking about this with respect to moving it to common_data, which would be easier for downstream to override via config.